dotnet/src/Connectors/Connectors.Memory.Kusto/KustoMemoryStore.cs

```diff
@@ -214,11 +218,14 @@ public async IAsyncEnumerable<string> GetCollectionsAsync([EnumeratorCancellatio
         {
             var key = reader.GetString(0);
             var metadata = reader.GetString(1);
-            var timestamp = !reader.IsDBNull(2) ? reader.GetString(2) : null;
+            DateTimeOffset? timestamp = !reader.IsDBNull(2) ? reader.GetDateTime(2) : null;
             var similarity = reader.GetDouble(3);
             var recordEmbedding = withEmbeddings ? reader.GetString(4) : default;
 
-            var kustoRecord = new KustoMemoryRecord(key, metadata, recordEmbedding, timestamp);
+            var deserializedMetadata = KustoSerializer.DeserializeMetadata(metadata);
+            var deserializedembedding = KustoSerializer.DeserializeEmbedding(recordEmbedding);
+
+            var kustoRecord = new KustoMemoryRecord(key, deserializedMetadata, deserializedembedding, timestamp);
```

Comment:
@mohammedtabish0 It would be nice to extract this code out into a method rather than have the duplicate implementations